includes/config.sh

```diff
@@ -1,311 +1,355 @@
+    fi
+    if [[ -n ${SERVER_NAME+x} ]]; then
+        echo "> Setting server name to '$SERVER_NAME'"
+        sed -E -i "s/ServerName=\"[^\"]*\"/ServerName=\"$SERVER_NAME\"/" "$GAME_SETTINGS_FILE"
+        if [[ "$SERVER_NAME" == *"###RANDOM###"* ]]; then
+            RAND_VALUE=$RANDOM
+            echo "> Found standard template, using random numbers in server name"
+            sed -E -i -e "s/###RANDOM###/$RAND_VALUE/g" "$GAME_SETTINGS_FILE"
+            echo "> Server name is now 'jammsen-docker-generated-$RAND_VALUE'"
         fi
-        if [[ ! -z ${USEAUTH+x} ]]; then
-            echo "> Setting bUseAuth to $USEAUTH"
-            sed -E -i "s/bUseAuth=[a-zA-Z]*/bUseAuth=$USEAUTH/" ${GAME_PATH}/Pal/Saved/Config/LinuxServer/PalWorldSettings.ini
+    fi
+    if [[ -n ${SERVER_DESCRIPTION+x} ]]; then
+        echo "> Setting server description to '$SERVER_DESCRIPTION'"
+        sed -E -i "s/ServerDescription=\"[^\"]*\"/ServerDescription=\"$SERVER_DESCRIPTION\"/" "$GAME_SETTINGS_FILE"
+    fi
+    if [[ -n ${ADMIN_PASSWORD+x} ]]; then
+        echo "> Setting server admin password to '$ADMIN_PASSWORD'"
+        sed -E -i "s/AdminPassword=\"[^\"]*\"/AdminPassword=\"$ADMIN_PASSWORD\"/" "$GAME_SETTINGS_FILE"
+    fi
+    if [[ -n ${SERVER_PASSWORD+x} ]]; then
+        echo "> Setting server password to '$SERVER_PASSWORD'"
+        sed -E -i "s/ServerPassword=\"[^\"]*\"/ServerPassword=\"$SERVER_PASSWORD\"/" "$GAME_SETTINGS_FILE"
+    fi
+    if [[ -n ${PUBLIC_PORT+x} ]]; then
+        echo "> Setting public port to '$PUBLIC_PORT'"
+        sed -E -i "s/PublicPort=[0-9]*/PublicPort=$PUBLIC_PORT/" "$GAME_SETTINGS_FILE"
+    fi
+    if [[ -n ${PUBLIC_IP+x} ]]; then
+        echo "> Setting public ip to '$PUBLIC_IP'"
+        sed -E -i "s/PublicIP=\"[^\"]*\"/PublicIP=\"$PUBLIC_IP\"/" "$GAME_SETTINGS_FILE"
+    fi
+    if [[ -n ${RCON_ENABLED+x} ]]; then
+        echo "> Setting rcon-enabled to '$RCON_ENABLED'"
+        sed -E -i "s/RCONEnabled=[a-zA-Z]*/RCONEnabled=$RCON_ENABLED/" "$GAME_SETTINGS_FILE"
+    fi
+    if [[ -n ${RCON_PORT+x} ]]; then
+        echo "> Setting RCONPort to '$RCON_PORT'"
+        sed -E -i "s/RCONPort=[0-9]*/RCONPort=$RCON_PORT/" "$GAME_SETTINGS_FILE"
+    fi
+    if [[ -n ${REGION+x} ]]; then
+        echo "> Setting Region to '$REGION'"
+        sed -E -i "s/Region=\"[^\"]*\"/Region=\"$REGION\"/" "$GAME_SETTINGS_FILE"
+    fi
+    if [[ -n ${USEAUTH+x} ]]; then
+        echo "> Setting bUseAuth to '$USEAUTH'"
+        sed -E -i "s/bUseAuth=[a-zA-Z]*/bUseAuth=$USEAUTH/" "$GAME_SETTINGS_FILE"
+    fi
+    if [[ -n ${BAN_LIST_URL+x} ]]; then
+        echo "> Setting BanListURL to '$BAN_LIST_URL'"
+        sed -E -i "s~BanListURL=\"[^\"]*\"~BanListURL=\"$BAN_LIST_URL\"~" "$GAME_SETTINGS_FILE"
+    fi
+
+    es ">> Finished setting up PalWorldSettings.ini\n"
+}
+
+function setup_rcon_config_file () {
+    ew ">> Setting up 'rcon.yaml' ...\n"
+
+    ew ">> RCON_ENABLED: ${RCON_ENABLED}\n"
```

Comment:
To much verboosity for doing 2 seds, need that more compact.

Pasting 8+ lines needs to add value and not just echos.